sklearn/ensemble/_forest.py

```diff
@@ -2985,3 +2985,12 @@ def transform(self, X):
         """
         check_is_fitted(self)
         return self.one_hot_encoder_.transform(self.apply(X))
+
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        tags._xfail_checks = {
+            "check_sample_weights_invariance": (
```

Comment:
I think if they're expected to be fixed, it'd be nice to add a `TODO(fixme)` kinda comment for these.

Reply:
You could even cross-link to the meta-issue URL (#16298) when we think the estimator check has revealed a bug.

For this specific estimator, however, I suspect that the problem is that `fit` is not deterministic and that the `assert_allclose` assertions in `check_sample_weights_invariance` are not expected to pass as such (we would need a statistical test instead of a deterministic invariance check`: we would need to run @snath-xoc's notebook on transformers to confirm whether or not this transformer has a sample weight problem or not.

Reply:
Happy to check this on transformers and update the gist

Reply:
And clustering models (in particular those that are sensitive to random inits such as k-means).